src/modules/SpaceCalculation.rb

```diff
@@ -1093,7 +1038,46 @@ def btrfs_used_size(directory)
     # where unit can be one of: "" (none) or "B", "KiB", "MiB", "GiB", "TiB", "PiB"
     # @return [Integer] size in bytes
     def size_from_string(size_str)
-      WFM.call("wrapper_storage", ["ClassicStringToByte", [size_str]])
+      classic_locale = true
+      # Assume bytes by default
+      size_str += "B" unless size_str =~ /[[:alpha:]]/
+      ::Storage.humanstring_to_byte(size_str, classic_locale)
+    end
+
+  private
+
+    # Filesystems to consider while checking the system available space
+    #
+    # @return [Array<Storage::Filesystem>]
+    def target_filesystems
+      filesystems = ::Storage::Filesystem.all(staging_devicegraph).to_a
+      filesystems.select! { |fs| fs.mountpoints.any? { |mp| mp.start_with?("/") } }
+      filesystems.reject! { |fs| TARGET_FS_TYPES_TO_IGNORE.include?(fs.type) }
+      filesystems
+    end
+
+    def staging_devicegraph
+      @storage_manager ||= Y2Packager::StorageManagerProxy.new
+      @storage_manager.staging
+    end
+
+    def filesystem_size(filesystem)
+      blk_device = filesystem.blk_devices[0]
+      # Only for local fs, NFS not supported yet in libstorage-ng
+      return 0 unless blk_device
+      blk_device.size
+    end
+
+    def filesystem_block_size(filesystem)
```

Comment:
The name of the function is confusing since it returns the block size of the underlying block device and not the block size of the filesystem. E.g. mkfs.ext4 also has a 'block-size' option.